BraceExpansion.java

```diff
@@ -0,0 +1,81 @@
+// Time Complexity : not sure what the worst case would be 
+// Space Complexity : maybe O(N), for max case recursion depth, keeping the sorted chars
```

Comment:
assume S is broke into blocks of avg length k
so at each point, k choices are available, number of blocks = n/k
time - O(k^(n/k))
space - O(n/k)